### PR TITLE
brag/client: use fsmap instead of mdsmap

### PR DESCRIPTION
**Pull request by zphj1987**
use fsmap instead of mdsmap
Signed-off-by: zphj1987  zphj1987@gmail.com

### CODE REVIEW

**Comment by zphj1987** [comment]
@tchaikov : would you have time to review this change ?

File: src/brag/client/ceph-brag [hunk 1]
```diff
@@ -258,7 +258,7 @@ def get_nums():
   num_mons = len(oj['monmap']['mons'])
   num_osds = int(oj['osdmap']['osdmap']['num_in_osds'])
   try:
-    num_mdss = oj['mdsmap']['in']
+    num_mdss = oj['fsmap']['in']
```
Comment (tchaikov) [inline src/brag/client/ceph-brag]: this is introduced by 4e9b953. @jcsp is it okay to use fsmap here? if yes, shall we backport this to jewel and kraken?

Comment (tchaikov) [inline src/brag/client/ceph-brag]: @zphj1987 could you fill more details in the commit message? the one in https://github.com/ceph/ceph/pull/13079/commits looks great, IMO.

**Comment by tchaikov** [comment]
@zphj1987 also, your signed-off-by line does not look right:
```
Signed-off-by: zphj1987  zphj1987@gmail.com
```

better off putting your full name here, and quote your mail with brackets, like:
```
Signed-off-by: Peng Zhang <zphj1987@gmail.com>
```

**Comment by zphj1987** [comment]
@tchaikov
thank you for your help , i will read more about how to write it

File: src/brag/client/ceph-brag (hunk 1 above)
Comment (jcsp) [inline src/brag/client/ceph-brag]: I'm not sure if this tool is supposed to be backwards compatible with older clusters -- if it is, then this code should check for both mdsmap and fsmap. the mdsmap->fsmap change happened in jewel.

Comment (jcsp) [inline src/brag/client/ceph-brag]: & yes, we should backport this to jewel and kraken once it's ready

**Comment by tchaikov** [comment]
@zphj1987 ping?

**Comment by tchaikov** [comment]
superseded by #13798

**Pull request closed without merging**